### PR TITLE
Encode by backend

### PR DESCRIPTION
**Pull request by nguyenhuudungz**

### CODE REVIEW

File: .env.production [hunk 1]
```diff
@@ -7,7 +7,8 @@ REACT_APP_MIGRATOR_ADDRESS=0x6a65e062ce8290007301296f3c6ae446af7bdeec
 REACT_APP_MAINNET_ENV=production
 REACT_APP_INFURA_KEY="9aa3d95b3bc440fa88ea12eaa4456161"
 
-REACT_APP_AGGREGATOR_API="https://aggregator-api.kyberswap.com"
+#REACT_APP_AGGREGATOR_API="https://aggregator-api.kyberswap.com"
+REACT_APP_AGGREGATOR_API="https://aggregator-api.dev.kyberengineering.io"
```
Comment (tiendv89) [inline .env.production:11]: Please double check this one

File: src/utils/aggregator.ts [hunk 2]
```diff
@@ -314,15 +321,24 @@ export class Aggregator {
...
     currencyAmountIn: CurrencyAmount,
     currencyOut: Currency,
     saveGas = false,
     dexes = '',
-    gasPrice?: GasPrice,
-    signal?: AbortSignal,
+    gasPrice: GasPrice | undefined,
+    slippageTolerance: number,
+    deadline: BigNumber | undefined,
+    to: string | undefined,
```
Comment (tiendv89) [inline src/utils/aggregator.ts:340]: You are missing the feeconfig params

File: .env.production (hunk 1 above)
Comment (nguyenhuudungz) [reply .env.production:11]: Will change to prod before merging.

File: src/utils/aggregator.ts (hunk 2 above)
Comment (nguyenhuudungz) [reply src/utils/aggregator.ts:342]: I added it.

File: .env.production (hunk 1 above)
Comment (nguyenhuudungz) [reply .env.production]: done

File: src/hooks/useSwapV2Callback.ts [hunk 3]
```diff
@@ -408,132 +455,160 @@ export function useSwapV2Callback(
...
+        },
+        gasEstimate,
+      } = successfulEstimation
+
+      console.log(
+        '[gas_price] swap used: ',
+        gasPrice?.standard ? `api/node: ${gasPrice?.standard} wei` : 'metamask default',
+      )
+
+      console.log(`\n\n********************\n\n`)
```
Comment (viet-nv) [inline src/hooks/useSwapV2Callback.ts:533]: remember to remove this

File: src/hooks/useSwapV2Callback.ts [hunk 4]
```diff
@@ -408,132 +455,160 @@ export function useSwapV2Callback(
...
 
-        console.log(
-          '[gas_price] swap used: ',
-          gasPrice?.standard ? `api/node: ${gasPrice?.standard} wei` : 'metamask default',
-        )
-        return contract[methodName](...args, {
-          gasLimit: calculateGasMargin(gasEstimate),
-          ...(gasPrice?.standard ? { gasPrice: ethers.utils.parseUnits(gasPrice?.standard, 'wei') } : {}),
-          ...(value && !isZero(value) ? { value, from: account } : { from: account }),
+      console.log(`sendTransactionOption`, sendTransactionOption)
```
Comment (viet-nv) [inline src/hooks/useSwapV2Callback.ts:591]: do we need this console.log?

File: src/state/swap/reducer.ts [hunk 5]
```diff
@@ -38,7 +38,20 @@ const initialState: SwapState = {
   },
   recipient: null,
   saveGas: false,
-  feeConfig: null,
+  feeConfig:
```
Comment (viet-nv) [inline src/state/swap/reducer.ts:41]: Remember to remove this when merge

File: src/state/swap/reducer.ts [hunk 6]
```diff
@@ -57,7 +70,9 @@ export default createReducer<SwapState>(initialState, builder =>
           typedValue: typedValue,
           recipient,
           saveGas: state.saveGas,
-          feeConfig,
+          // TODO (NHDz):
```
Comment (viet-nv) [inline src/state/swap/reducer.ts:73]: same as above

File: src/utils/fee.ts [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+import { BIPS_BASE, RESERVE_USD_DECIMALS } from 'constants/index'
+import { Aggregator } from 'utils/aggregator'
+import { formattedNum } from 'utils/index'
+import { parseUnits } from 'ethers/lib/utils'
+
+// This function is not correct, the result will be rounded.
+// Eg. 0.9999 (amountIn) * 0.0008 (fee bps currency_in) = 0.000799 (round 6 number, for example, swap from usdt)
+// => amount without fee in = 0.9999 - 0.000799 = 0.999101
+// We have amountPlusFee = 0.999101 / (1 - 0.0008) = 0.9999009207 => Wrong.
+// TODO: Delete this function and logic of encoding in frontend after releasing it in backend.
```
Comment (viet-nv) [inline src/utils/fee.ts:12]: remember this please

**Review by viet-nv — CHANGES_REQUESTED** [review]

**Pull request closed without merging**